rkt/metadata_service.go

```diff
@@ -541,12 +546,20 @@ func runMetadataService(args []string) (exit int) {
 
 	log.Print("Metadata service running...")
 
-	if err = srv.Serve(l); err != nil {
-		log.Printf("Error serving HTTP: %v", err)
+	go func() {
+		if err = srv.Serve(unixl); err != nil {
+			stderr("Error serving registration HTTP: %v", err)
+			cleanup()
+			os.Exit(1)
+		}
+	}()
+
+	if err = srv.Serve(tcpl); err != nil {
```

Comment:
Well, if the PR is coming Real Soon then that's fine, but I don't want it
to block 0.5.0 in the next ~day..

On Mon, Mar 23, 2015 at 4:33 PM, Eugene Yakubovich <notifications@github.com
> wrote:

> In rkt/metadata_service.go
> <https://github.com/coreos/rocket/pull/637#discussion_r26991534>:
>
> > @@ -541,12 +546,20 @@ func runMetadataService(args []string) (exit int) {
> >
> >  	log.Print("Metadata service running...")
> >
> > -	if err = srv.Serve(l); err != nil {
> > -		log.Printf("Error serving HTTP: %v", err)
> > +	go func() {
> > +		if err = srv.Serve(unixl); err != nil {
> > +			stderr("Error serving registration HTTP: %v", err)
> > +			cleanup()
> > +			os.Exit(1)
> > +		}
> > +	}()
> > +
> > +	if err = srv.Serve(tcpl); err != nil {
>
> Ok, I'll do that. I was going to redo all this in a separate PR that also
> handles signals correctly but I can do some sync part here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/rocket/pull/637/files#r26991534>.
>